### PR TITLE
Azure automated deployment for OPEA applications - Infosys

### PR DESCRIPTION
**Pull request by kkrishTa**
## Description

Infosys contribution to OPEA.
Azure AKS terraforms and steps to execute to deploy OPEA applications.

## Issues

NA

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

None.

## Tests

Tested terraforms on Azure platform.

### CODE REVIEW

File: cloud-service-provider/azure/aks/terraform/README.md
Comment (arun-gupta) [inline cloud-service-provider/azure/aks/terraform/README.md:3]: `ASK` -> `AKS`

File: cloud-service-provider/azure/aks/terraform/README.md
Comment (arun-gupta) [inline cloud-service-provider/azure/aks/terraform/README.md:7]: AKS

File: cloud-service-provider/azure/aks/terraform/README.md
Comment (arun-gupta) [inline cloud-service-provider/azure/aks/terraform/README.md:13]: AKS - this is a global search and replace through out this doc

File: cloud-service-provider/azure/aks/terraform/README.md
Comment (arun-gupta) [inline cloud-service-provider/azure/aks/terraform/README.md:16]: Do we need this autoscaling to 10 nodes?

File: cloud-service-provider/azure/aks/terraform/README.md
Comment (arun-gupta) [inline cloud-service-provider/azure/aks/terraform/README.md:15]: Do we need a SPOT instance? Can this be a standard virtual machine?

File: cloud-service-provider/azure/aks/terraform/README.md
Comment (arun-gupta) [inline cloud-service-provider/azure/aks/terraform/README.md:19]: Not sure what needs to be done here?

File: cloud-service-provider/azure/aks/terraform/README.md
Comment (arun-gupta) [inline cloud-service-provider/azure/aks/terraform/README.md:67]: It would be useful to state that "ensure all pods are running or show 1/1 or something similar"

**Review by arun-gupta — CHANGES_REQUESTED** [review]
Thanks for the PR.

I did not try out the instructions but leaving comments based upon reading.

File: cloud-service-provider/azure/aks/terraform/README.md
Comment (kkrishTa) [reply cloud-service-provider/azure/aks/terraform/README.md:15]: To support cost optimization, we have mentioned SPOT. Since instance is a variable, we will mention standard/SPOT in the readme file and mention about cost optimization in the variable file.

File: cloud-service-provider/azure/aks/terraform/README.md
Comment (kkrishTa) [reply cloud-service-provider/azure/aks/terraform/README.md:67]: Sure Will add the statement.

File: cloud-service-provider/azure/aks/terraform/README.md
Comment (poussa) [reply cloud-service-provider/azure/aks/terraform/README.md:16]: I think it is reasonable number.

**Review by poussa — APPROVED** [review]
LGTM. Just change the cluster name from opea-chatqna to opea.

File: cloud-service-provider/azure/aks/terraform/opea-chatqna.tfvars
Comment (poussa) [inline cloud-service-provider/azure/aks/terraform/opea-chatqna.tfvars:1]: let's use cluster name `opea`. The AWS/EKS cluster will change to use that name as well.

Comment (kkrishTa) [reply cloud-service-provider/azure/aks/terraform/opea-chatqna.tfvars]: Updated cluster name to `opea`

**Comment by kkrishTa** [comment]
Hello @poussa ,
Review comments are fixed. Could you please approve and merge the code?

**Comment by poussa** [comment]
> Hello @poussa , Review comments are fixed. Could you please approve and merge the code?

I have already approved. We need a 2nd approver. I'll work on that now.

**Comment by poussa** [comment]
@kkrishTa can you also fix the DCO. That is, add signed-off-by line to the commit (-s).

**Comment by kkrishTa** [comment]
> @kkrishTa can you also fix the DCO. That is, add signed-off-by line to the commit (-s).

@poussa Sure. Fixed this one.

**Review by lianhao — APPROVED** [review]

**Comment by kkrishTa** [comment]
Hello @poussa / @lianhao ,
The pipeline got failed with an error: 
`/home/runner/work/GenAIInfra/GenAIInfra/docs/_build/rst/GenAIInfra/cloud-service-provider/azure/aks/terraform/README.md: WARNING: document isn't included in any toctree`
 
But I see that ".md files not in toc tree" is ignored as per https://github.com/opea-project/docs/blob/main/.known-issues/sphinx.conf. Though this is a warning, the pipeline step has failed.

Could you please advise fix on this?

**Comment by lianhao** [comment]
> 
> Could you please advise fix on this?
@poussa will create another PR to add the doc link to those exiting md files under cloud-service-provider directory, then you can add the link follow his convention. 

It's a newly added doc CI check complaining.

**Comment by poussa** [comment]
@kkrishTa once this [PR#633](https://github.com/opea-project/GenAIInfra/pull/633) is merged, you can update yours to add link to Azure/AKS readme. Then the ci/cd is happy and we can merge.

**Comment by poussa** [comment]
@kkrishTa can you resolve the conflict, please?

**Comment by poussa** [comment]
The readme filename is `README.md` not `README.MD` as it is in the link now. That's why ci/cd is still failing.

One more fix and we are there...

**Comment by lianhao** [comment]
The doc team is investigating why the `Check Online Document Building` CI fails now. Let's hold on this until they figure out the solution.

**Review by mkbhanda — APPROVED** [review]
LGTM

**Comment by mkbhanda** [comment]
Let me find someone on the docs team to help on this @kkrishTa. You have been most patient with our CICD tests.

**Comment by lianhao** [comment]
@kkrishTa The doc teams has revolved the doc build failure. For the `Check Paths and Hyperlinks / check-the-validity-of-hyperlinks-in-README` issue, could you please use relative link instead, just like PR #634

**Comment by kkrishTa** [comment]
> @kkrishTa The doc teams has revolved the doc build failure. For the `Check Paths and Hyperlinks / check-the-validity-of-hyperlinks-in-README` issue, could you please use relative link instead, just like PR #634

sure @lianhao . Once this is merged, I will pull the latest and update the link apporpriately.

**Comment by lianhao** [comment]
> > @kkrishTa The doc teams has revolved the doc build failure. For the `Check Paths and Hyperlinks / check-the-validity-of-hyperlinks-in-README` issue, could you please use relative link instead, just like PR #634
> 
> sure @lianhao . Once this is merged, I will pull the latest and update the link apporpriately.

No need to wait for PR #634, just go ahead to fix your PR by using relative links to newly added .md files

**Comment by kkrishTa** [comment]
> > > @kkrishTa The doc teams has revolved the doc build failure. For the `Check Paths and Hyperlinks / check-the-validity-of-hyperlinks-in-README` issue, could you please use relative link instead, just like PR #634
> > 
> > 
> > sure @lianhao . Once this is merged, I will pull the latest and update the link apporpriately.
> 
> No need to wait for PR #634, just go ahead to fix your PR by using relative links to newly added .md files. Let's get your PR to land-in first.

Updated readme. Since it has common read me, upadted other relative links as in #634 without breaking AWS readme link.

**Review by NeoZhangJianyu — APPROVED** [review]

**Pull request merged**
